### PR TITLE
RDM-3144 CCD displays 3 events all called ‘Apply for a divorce’ when logged in with solicitor account

### PR DESCRIPTION
**Pull request by tzarsmango**
The `/jurisdictions` api now returns the ACLs for the caseTypes, case events and case states. Added necessary authorisation checks to filter out unauthorised items.

There is a relevant ccd-definition-store-api [PR](https://github.com/hmcts/ccd-definition-store-api/pull/186) which must be merged and deployed to production before this PR.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/AuthorisedGetUserProfileOperation.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    @Override
+    public UserProfile execute(final String userToken) {
+        return filterCaseTypes(getUserProfileOperation.execute(userToken));
+    }
+
+    private UserProfile filterCaseTypes(UserProfile userProfile) {
+        final Set<String> userRoles = getUserRoles();
+        Arrays.stream(userProfile.getJurisdictions()).forEach(
+            jurisdictions -> jurisdictions.setCaseTypes(
```
Comment (vlaurin) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/AuthorisedGetUserProfileOperation.java]: `jurisdictions` should be singular

File: src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/AuthorisedGetUserProfileOperation.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                jurisdictions.getCaseTypes()
+                    .stream()
+                    .map(caseType -> verifyAccess(caseType, userRoles, CAN_READ))
```
Comment (vlaurin) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/AuthorisedGetUserProfileOperation.java]: Here we are testing if the user `CAN_READ`, however, this is not relevant if the case types or events are used for creating cases as you could have READ write on case type or event, but be missing the Create right.

File: src/test/java/uk/gov/hmcts/ccd/domain/service/aggregated/AuthorisedGetUserProfileOperationTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
+package uk.gov.hmcts.ccd.domain.service.aggregated;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.*;
```
[the remark this answers is not in the document]
Comment (tzarsmango) [reply src/test/java/uk/gov/hmcts/ccd/domain/service/aggregated/AuthorisedGetUserProfileOperationTest.java]: changed the import grouping settings in intelliJ though this Codacy error doesn't make sense

File: src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/AuthorisedGetUserProfileOperation.java (hunk 1 above)
Comment (tzarsmango) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/AuthorisedGetUserProfileOperation.java]: done

File: src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/AuthorisedGetUserProfileOperation.java (hunk 2 above)
Comment (tzarsmango) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/AuthorisedGetUserProfileOperation.java]: updated the Access Type to CAN_CREATE

File: src/main/java/uk/gov/hmcts/ccd/endpoint/ui/QueryEndpoint.java [hunk 4]
```diff
@@ -106,6 +99,21 @@ public QueryEndpoint(
...
     }
 
+    @RequestMapping(value = "/caseworkers/{uid}/jurisdictions", method = RequestMethod.GET)
+    @ApiOperation(value = "Get jurisdictions available to the user")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "List of jurisdictions for the given access criteria"),
+        @ApiResponse(code = 404, message = "No jurisdictions found for given access criteria")})
+    public List<JurisdictionDisplayProperties> getJurisdictions(@RequestParam(value = "access") String access) {
+        List<JurisdictionDisplayProperties> jurisdictions = Arrays.asList(
+            getUserProfileOperation.execute(accessMap.get(access)).getJurisdictions());
```
Comment (vlaurin) [inline src/main/java/uk/gov/hmcts/ccd/endpoint/ui/QueryEndpoint.java]: Missing validation of the `access` property?

Comment (tzarsmango) [reply src/main/java/uk/gov/hmcts/ccd/endpoint/ui/QueryEndpoint.java]: added validation

**Pull request merged**